### ISSUE TITLE
Settings for installer modes

### ISSUE THREAD

**Issue by denelon** [microsoft/winget-cli#910]
# Description of the new feature/enhancement

Users should be able to specify a requirement or a preference for install modes in their settings.

Some installers do not support "silent" or unattended install. This can have a negative impact on scripted installs or import. Some users may wish to have "interactive" as their preferred installation mode.

# Proposed technical implementation details (optional)

If a user requires "silent" as their install mode, search should not show results for installers that do not support "silent" install.
if a user attempts to install a package, and their install mode requirement cannot be respected, the user should be informed.
If a user attempts to import a set of packages, their install mode preferences should be respected.
If a user attempts to import a set of packages, and their install mode requirement cannot be respected, the user should be informed.

**Comment by JustMyGithub** [comment]
Can winget handle command switches for installers already? Most silent setups I know are not silent themselves, but require cli parameters like "setup.exe --silent" and I have not read winget being able to handle that yet.

**Comment by denelon** [comment]
@JustMyGithub yes. If the installer is MSIX, MSI, or an installer type we recognize like Nullsoft, Inno, etc. the client knows what switches to pass, so they are not needed in the manifest. If it is a custom .exe installer you would pass the switches to the installer via the manifest in the "InstallerSwitches" key.

```JSON
"InstallerSwitches": {
      "type": "object",
      "properties": {
        "Silent": {
          "type": [ "string", "null" ],
          "minLength": 1,
          "maxLength": 512,
          "description": "Silent is the value that should be passed to the installer when user chooses a silent or quiet install"        },
```

**Comment by JustMyGithub** [comment]
So some tools show the setup wizard when installed with Winget and other dont? Thats a pretty unintuitive user experience.

Is "null" in your example the place where the cli parameters would be configured? (like 'type": [ "string", "--silent" ]') ?

**Comment by denelon** [comment]
No, that code snippet is from the JSON schema designed to aid building a manifest in an IDE like Visual Studio Code with JSON schema validation support. The node is nullable, but our "business logic" for the community repository requires the Silent and SilentWithProgress switches for some .exe installers.

If you were to host a private repository, or use a local manifest from your file system, the InstallerSwitches node is not required.

**Comment by hansmbakker** [comment]
I added the requirement for `interactive` installation [in my manifest PR](https://github.com/microsoft/winget-pkgs/pull/68036/files#diff-cf47f1c77135c03f353b28417d07c4a737506c91294abaa585e4fc1bb4e7d9b3) but 
```
winget install --manifest .\manifests\a\Audient\Sono\4.2.1.0\
```
does not trigger an interactive installation - is that related to this issue?

**Comment by denelon** [comment]
There are already options for the installer behavior.

winget install -?

You can use "-i, or --interactive", or "-h or --silent" to specify interactive or silent behavior on the command line.

We do need to add settings for the default preferred behavior in the settings.json file.

The default experience is "silent with progress" meaning the installer can pop UI, but the install must be able to complete without any user interaction.

I think we'd need to add something for the default so if a user had a setting to prefer or require one of the other install modes, it could be overridden with an argument on the command line.

**Comment by denelon** [comment]
Consolidating this into #823, which covers the same capability (filtering/selecting packages by installer mode). I've expanded #823's scope to incorporate the details here - settings + command-line requirement/preference, search filtering, install-time notification when a required mode can't be honored, and import behavior - so nothing is lost.

Closing as a duplicate of #823; please follow that issue for updates.

**If you're subscribed here, please head over to #823 and add a 👍 reaction to the original post** so your interest is counted toward that issue. Reactions - not comments - are how we gauge demand, and closing this duplicate would otherwise lose your signal.

_(Drafted with GitHub Copilot assistance.)_
